src/table.rs

```diff
@@ -61,40 +62,351 @@ impl Table {
+                let card1 = *deck_iterator.next().unwrap();
+                let card2 = *deck_iterator.next().unwrap();
+                player.deal([card1, card2]);
+                self.pot += player.bet(self.ante);
+                // the ante doesn't count as a turn so clarify the bot hasn't had a turn
+                player.has_had_turn_this_round = false;
+            }
+        }
+    }
+
+    pub fn get_state_string_for_player(&self, id: i8) -> JsonValue {
+        let player_strings: Vec<_> = self.players.iter().map(|x| if x.get_id() == id { x.as_json() } else { x.as_json_no_secret_data() }).collect();
+        object! {
+            flop: self.get_flop_string_secret(),
+            turn: self.get_turn_string_secret(),
+            river: self.get_river_string_secret(),
+            dealer_button_index: self.dealer_button_index,
+            players: player_strings,
+            hand_number: self.hand_number,
+        }
+    }
+    fn is_betting_over(&self) -> bool {
+        let all_players_bet_or_folded = self.check_all_players_ready_for_next_round();
+        let all_players_equal_bet = self.check_all_active_players_same_bet();
+        all_players_bet_or_folded && all_players_equal_bet
+    }
+    fn get_max_bet(&self) -> i32 {
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { 0 }
+            PlayerState::Active(active_state) => { active_state.current_bet }
+        }).max().unwrap()
+    }
+    fn check_all_players_ready_for_next_round(&self) -> bool {
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { true }
+            PlayerState::Active(_) => { x.has_had_turn_this_round }
+        }).reduce(|x, y| x && y).unwrap()
+    }
+    fn check_all_active_players_same_bet(&self) -> bool {
+        let max_bet = self.get_max_bet();
+        self.players.iter().map(|x| match x.player_state {
+            PlayerState::Folded => { true }
+            PlayerState::Active(a) => {
+                x.total_money < max_bet || a.current_bet == max_bet
+            }
+        }
+        ).reduce(|x, y| x && y).unwrap()
+    }
+    fn get_alive_player_count(&self) -> usize {
+        self
+            .players
+            .iter()
+            .map(|x| match x.player_state {
+                PlayerState::Folded => { 0 }
+                PlayerState::Active(_) => { 1 }
+            })
+            .reduce(|x, y| x + y)
+            .unwrap()
+    }
+    fn pick_winner(&mut self) {
+        // This is the everyone but one person has folded case, give that person the winnings
+        if self.get_alive_player_count() == 1 {
+            self.players.iter_mut().find(|x| match x.player_state {
+                PlayerState::Folded => { false }
+                PlayerState::Active(_) => { true }
+            }).unwrap().total_money += self.pot;
+        }
+        // TODO give the winnings out based on hand strength
+        self.deal();
```

Comment:
Picky, but dealing doesn't really fit in this function. Could rename from pick_winner or just call deal after.

Reply:
Function name changed and added a doc string